### PR TITLE
file changes

### PR DESCRIPTION
**Pull request by annamharikrishna**
1.uses requirejs for viewer only
2.file name changed
3.to view changes run viewer.html

### CODE REVIEW

**Comment by mithuntnt** [comment]
File names of classes to be UpperCamelCase.js  Ex: Cabinet js.
Pull from the branch before giving pull request.
Remove the .idea files from commit.
There is a git rename option which is preferred to delete and add.

File: cabinet-studio/js/cabinetviewer/3DObjectCreator.js [hunk 1]
```diff
@@ -1,41 +1,42 @@
-(function(cs){
+define(["cabinetviewer/Viewer"], function (cs) {
     "use strict";
-    cs.To3DObject = function(plankData){
+    cs.To3DObject = function (plankData) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/3DObjectCreator.js]: function should define to what object it is generating 3DObject

Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/3DObjectCreator.js]: "plank" is better than plankData

File: cabinet-studio/js/cabinetviewer/3DObjectCreator.js [hunk 2]
```diff
@@ -1,41 +1,42 @@
...
         var plankGeometry = new THREE.Geometry();
         var input = plankData.getVertices();
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/3DObjectCreator.js]: "input" is not defining what it is containing

File: cabinet-studio/js/cabinetviewer/3DObjectCreator.js [hunk 3]
```diff
@@ -1,41 +1,42 @@
...
-        for(var i=0;i<input.length;i++){
-            plankGeometry.vertices.push(new THREE.Vector3(input[i].x,input[i].y,input[i].z));
+        for (var i = 0; i < input.length; i++) {
+
+            plankGeometry.vertices.push(new THREE.Vector3(input[i].x, input[i].y, input[i].z));
         }
 
-        var materialLaminate = new THREE.MeshBasicMaterial({map:THREE.ImageUtils.loadTexture('images/texture3.jpg')});
-        var materialedge = new THREE.MeshBasicMaterial({map:THREE.ImageUtils.loadTexture('images/texture4.jpg')});
+        var materialLaminate = new THREE.MeshBasicMaterial({map: THREE.ImageUtils.loadTexture('images/texture3.jpg')});
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/3DObjectCreator.js]: No one knows whether texture3.jpg belongs to lamination or edgebinding,etc. Maintain data struc for that

File: cabinet-studio/js/cabinetviewer/3DObjectCreator.js [hunk 4]
```diff
@@ -45,58 +46,57 @@
         var textureheight = 150;
 
         var front = [
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/3DObjectCreator.js]: front means?

File: cabinet-studio/js/cabinetviewer/config.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+define(["cabinetviewer/Viewer", "cabinetviewer/Storage"], function (c, storage) {
+
+    "use strict";
+
+    c.EditorConfig = {
+
+        name: "CFurnish_Editor",
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/config.js]: CFurnish_Editor means?

File: cabinet-studio/js/cabinetviewer/config.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+        this.data = c.EditorConfig;
+
+    };
+
+    c.Config.prototype = {
+
+        get: function (key) {
+
+            if (c.IsStorageAvailable) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/config.js]: is IsStorageAvailable a variable? wer is it defined

File: cabinet-studio/js/cabinetviewer/editor.js [hunk 7]
```diff
@@ -48,7 +43,6 @@
 
             this.scene.add(object);
 
-
         },
 
         moveObject: function (object, parent, before) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/editor.js]: provide description for this

File: cabinet-studio/js/cabinetviewer/Utils.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        checkWebgl: function (success) {
+
+            if (Modernizr.webgl) {
+
+                success();
+
+            }
+        },
+
+        FeetToCm: function (feet) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/Utils.js]: naming conv

File: cabinet-studio/js/cabinetviewer/Utils.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+            return parseFloat((feet / 0.032808).toFixed(2));
+
+        },
+
+        InchToCm: function (inches) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/Utils.js]: naming conv

File: cabinet-studio/js/cabinetviewer/Utils.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+            return parseFloat((inches / 2.54).toFixed(2));
+
+        },
+
+        CanvasVisibleDimensions: function (camera) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/Utils.js]: naming conv

File: cabinet-studio/js/cabinetviewer/Viewer.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+define(["core"], function (studio) {
```
Comment (abhinavvarma) [inline cabinet-studio/js/cabinetviewer/Viewer.js]: file name should be same as class name

Comment (annamharikrishna) [inline cabinet-studio/js/cabinetviewer/Viewer.js]: here studio.viewer is namespace

**Pull request merged**